pyterrier/pipelines.py

```diff
@@ -214,6 +223,7 @@ def Experiment(
             Applying a batch_size is useful if you have large numbers of topics, and/or if your pipeline requires large amounts of temporary memory
             during a run.
         drop_unused(bool): If True, will drop topics from the topics dataframe that have qids not appearing in the qrels dataframe. 
+        filter_qrels(bool): If True, will drop topics from the qrels dataframe that have qids not appearing in the topics dataframe. 
```

Comment:
Yeah, you can have both true. It's essentially left outer join, right outer join, and inner join.

If qrels include A, B and topics include B, C:
 - `drop_unused=True filter_qrels=True` gives B
 - `drop_unused=True filter_qrels=False` gives A, B
 - `drop_unused=False filter_qrels=True` gives B, C
 - `drop_unused=False filter_qrels=False` gives A, B, C

I'm in favour of giving them more similar names. `filter_by_topics` (formerly `filter_qrels`) and `filter_by_qrels` (formerly `drop_unused`) sound reasonable to me. Should probably pull from kwargs and give a warning if `drop_unused` is used for backward compatibility.

Reply:
Agreed. We may need something in the documentation about this filtering.

Are we sure 
 * `filter_by='topics'`
 * `filter_by='qrels'`
 * `filter_by='both'`
 * `filter_by=None`

shouldnt be the correct nomenclature?

Reply:
I've been thinking about it in terms of separate decisions the user could make, and as a result, a single option that does both could be confusing.

1. Do you want skip topics that do not appear in the qrels? If so, set `filter_by_qrels=True`.
2. Do you want to evaluate across all topics from the qrels, even if they do not appear in the requested topics? If so, set `filter_by_topics=False`. I'd expect this to be particularly uncommon and be limited to situations like the one brought up in #226.

I could probably be convinced otherwise, though.

Reply:
Ok, you convinced me. Carry on.